src/components/side_panel/data_validation/data_validation_panel.ts

```diff
@@ -19,6 +20,21 @@ export class DataValidationPanel extends Component<Props, SpreadsheetChildEnv> {
 
   state = useState<State>({ mode: "list", activeRule: undefined });
 
+  setup(): void {
+    if (this.props.ruleID) {
+      this.onPreviewClick(this.props.ruleID);
+    }
+    onWillUpdateProps((nextProps: Props) => {
+      if (nextProps.ruleID !== this.props.ruleID) {
+        if (nextProps.ruleID) {
+          this.onPreviewClick(nextProps.ruleID);
+        } else {
+          this.onExitEditMode();
```

Comment:
What about adding another entry in the side panel registry to be able to open the DV "editor" directly ?
`onWillUpdateProps` are always smelly ;)
It would also keep this component simpler

Reply:
Done :)